components/RenderProposalContentLambda.tsx

```diff
@@ -214,18 +230,41 @@ export const contentToData = (
         addresses: undefined,
         entrypoints: undefined,
         params: JSON.stringify(
-          lambdaData[0].txs.map(({ to_, token_id, amount }) => ({
-            fa2_address:
-              token?.token.contract.address ?? lambda?.contractAddress,
-            name: token?.token.metadata.name,
-            token_id,
-            to: to_,
-            amount: !!token?.token.metadata.decimals
-              ? BigNumber(amount)
-                  .div(BigNumber(10).pow(token?.token.metadata.decimals ?? 0))
-                  .toString()
-              : amount.toString() + "*",
-          }))
+          lambdaData[0].txs.map(({ to_, token_id, amount }) => {
+            const token = walletTokens.find(
+              token =>
+                token.token.contract.address === lambda?.contractAddress &&
+                token.token.tokenId === token_id.toString()
+            );
+            const metadata = token?.token.metadata;
+            let imageUri = metadata?.thumbnailUri;
+
+            if (
+              imageUri === undefined &&
+              metadata !== undefined &&
+              "displayUri" in metadata &&
+              metadata.displayUri !== undefined
+            )
+              imageUri = metadata.displayUri;
+
+            imageUri = imageUri?.includes("ipfs")
+              ? `${THUMBNAIL_URL}/${imageUri.replace("ipfs://", "")}`
+              : imageUri;
```

Comment:
It's the 2nd time is this code, refactoring it into a function would be a good idea I think

Reply:
moved to util function